x-pack/plugins/apm/public/hooks/use_fetcher.tsx

```diff
@@ -130,7 +148,7 @@ export function useFetcher<TReturn>(
     doFetch();
 
     return () => {
-      didCancel = true;
+      controller.abort();
```

Comment:
What will the `FETCH_STATUS` state be when a request is aborted? Should we introduce a new state or is that overkill (I'd rather not if we can avoid it).

Reply:
Aborted is not necessarily a state here, it can be a result of a new request, the component unmounting, or the useFetcher cb not returning a promise. We were already "soft-canceling" requests before, so I don't think this change should affect state but I'll think about it for a bit.

Reply:
Okay. I was trying to think through if we still needed the `if (!signal.aborted) {` checks. Previously, if useFetcher props changed, we used `didCancel = true` to avoid updating the state when the underlying request (which didn't get cancelled) returned.
Since we are now actually cancelling the request, will useFetcher re-render afterwards?

Reply:
We might not need the first one. But the second one we do, as the promise will be rejected when the request is aborted.